### ISSUE TITLE
Some questions about the code of the svm prediction part

### ISSUE THREAD

**Issue by anbo1024** [AFAgarap/cnn-svm#27]
This is the predicted output part of the svm network：
**output = tf.identity(tf.sign(output), name='prediction')
correct_prediction = tf.equal(tf.argmax(output, 1), tf.argmax(y_input, 1))**

After the output is processed by the function **tf.sign()**，a two-category tag that is processed into (1,-1).This is not predictable for multi-class handwritten data sets.
I want to ask you this question.

**Comment by AFAgarap** [comment]
@anbo1024 it can be used in a one-versus-all manner, i.e. in a one-hot encoded vector, instead of using 0 for the classes that a sample does not belong, we use -1 : `[0, 1, 0, 0] --> [-1, 1, -1, 1]`

**Comment by anbo1024** [comment]
@AFAgarap I still have some doubts.
1、You will [0, 1, 0, 0] --> [-1, 1, -1, 1]，Softmax regards the maximum subscript as the final classification result. How do you determine the final classification result with this encoding method?
2、I think that in CNN, whether using SVM or Softmax, the network output has been classified. SVM and Softmax calculate loss in different ways.

**output = tf.identity(tf.sign(output), name='prediction')**
**output = tf.identity((output), name='prediction')**
correct_prediction = tf.equal(tf.argmax(output, 1), tf.argmax(y_input, 1))

I think **tf. sign** and **tf. nn. softmax** functions can be removed.The following **Correct_prediction** function already contains the maximum operation.
I hope I can get your advice. Thank you very much.

**Comment by AFAgarap** [comment]
Well, in terms of having a one-hot encoded vector with a -1 in place of 0, e.g. `[-1, 1, -1, -1]`, which one is greater, -1 or 1? 1, yes? So, it's still fine to be there. But you're right, it can be discarded for the purposes of getting the training accuracy.

**Comment by anbo1024** [comment]
@AFAgarap I understand what you mean. Now, I use SVM as a loss function on my own dataset, but the performance is very poor, and Softmax performs very well. Why is this? Is the data still to be preprocessed?

**Comment by AFAgarap** [comment]
In Yichuan Tang's paper, _Deep Learning using Linear Support Vector Machines_, he used PCA and added Gaussian noise for MNIST. But that was on a feed-forward neural network with 2 layers having 512 units each.

**Comment by AFAgarap** [comment]
The paper is [here](https://arxiv.org/pdf/1306.0239.pdf)

**Comment by anbo1024** [comment]
Yes, I have read this article. Good results have been achieved in your code. Did you preprocess it?

**Comment by AFAgarap** [comment]
No, I didn't
